locations/spiders/gelsons.py

```diff
@@ -0,0 +1,71 @@
+import json
+import scrapy
+import re
+
+from locations.items import GeojsonPointItem
+from locations.hours import OpeningHours, DAYS
+
+
+class GelsonsSpider(scrapy.Spider):
+    name = "gelsons"
+    item_attributes = {"brand": "Gelson's Markets", "brand_wikidata": "Q16993993"}
+    allowed_domains = ["gelsons.com"]
+    download_delay = 0.2
+    start_urls = ("https://www.gelsons.com/sitemap.xml",)
+
+    def parse(self, response):
+        response.selector.remove_namespaces()
+        urls = response.xpath("//url/loc/text()").extract()
+        for url in urls:
+            if re.match(r".*stores/.+$", url) and "virtual" not in url:
+
+                yield scrapy.Request(
+                    url=url, callback=self.parse_store, meta={"url": url}
+                )
```

Comment:
This should be a SitemapSpider, you may need to use `sitemap_filter` to remove the virtual store. See [HalifaxGB](https://github.com/alltheplaces/alltheplaces/blob/0562b1ccd19d41f9fa835a20bd262a2173ecb2e8/locations/spiders/halifax_gb.py) as an example.

Reply:
Great, thanks! Not sure how I missed this class, given how many sitemap-based spiders I've written. 😅